osgp/platform/osgp-adapter-ws-smartmetering/src/test/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/ActualPowerQualityResponseMappingTest.java

```diff
@@ -0,0 +1,134 @@
+        assertThat(actualCaptureObject.getDataIndex()).as(MAPPED_FIELD_VALUE_MESSAGE)
+                                                      .isEqualTo(sourceCaptureObject.getDataIndex());
+    }
+
+    private void assertCaptureObjects(
+            final List<org.opensmartgridplatform.adapter.ws.schema.smartmetering.common.CaptureObject> actualCaptureObjects,
+            final List<CaptureObject> sourceCaptureObjects) throws AssertionError {
+
+        assertThat(actualCaptureObjects.size()).as(MAPPED_LIST_SIZE_MESSAGE).isEqualTo(sourceCaptureObjects.size());
+        for (int i = 0; i < actualCaptureObjects.size(); i++) {
+            this.assertCaptureObject(actualCaptureObjects.get(i), sourceCaptureObjects.get(i));
+        }
+    }
+
+    private void assertActualValues(
+            final List<org.opensmartgridplatform.adapter.ws.schema.smartmetering.monitoring.ActualValue> actualValues,
+            final List<ActualValue> sourceActualValues) throws AssertionError {
+
+        assertThat(actualValues.size()).as(MAPPED_LIST_SIZE_MESSAGE).isEqualTo(sourceActualValues.size());
+        for (int i = 0; i < actualValues.size(); i++) {
+            this.assertActualValue(actualValues.get(i), sourceActualValues.get(i));
+        }
+    }
+
+    private void assertActualValue(
+            final org.opensmartgridplatform.adapter.ws.schema.smartmetering.monitoring.ActualValue actualValue,
+            final ActualValue sourceActualValue) throws AssertionError {
+
+        final Object actual = actualValue.getStringValueOrDateValueOrFloatValue().get(0);
+        if (actual instanceof XMLGregorianCalendar) {
+            final Date actualDate = ((XMLGregorianCalendar) actual).toGregorianCalendar().getTime();
+            assertThat(actualDate).as(MAPPED_FIELD_VALUE_MESSAGE).isEqualTo(sourceActualValue.getValue());
+        } else {
+            assertThat(actual).as(MAPPED_FIELD_VALUE_MESSAGE).isEqualTo(sourceActualValue.getValue());
+        }
+    }
+
+    private List<CaptureObject> makeCaptureObjects() {
+        final List<CaptureObject> captureObjects = new ArrayList<>();
+        captureObjects.add(new CaptureObject(8L, "0.0.1.0.0.255", 2, 0, OsgpUnitType.UNDEFINED.name()));
+        captureObjects.add(new CaptureObject(3L, "1.1.1.0.0.255", 2, 0, OsgpUnitType.W.name()));
+        captureObjects.add(new CaptureObject(3L, "2.2.2.0.0.255", 2, 0, OsgpUnitType.W.name()));
+        captureObjects.add(new CaptureObject(4L, "3.3.3.0.0.255", 2, 0, OsgpUnitType.W.name()));
+        return captureObjects;
+    }
+
+    private List<ActualValue> makeActualValues() {
+        return this.makeActualValue(new DateTime(2017, 1, 1, 1, 0, 0, DateTimeZone.UTC).toDate(), "test1",
+                new BigDecimal(1.1d), 111L);
+    }
+
+    private List<ActualValue> makeActualValue(final Serializable... values) {
+        final List<ActualValue> actualValues = new ArrayList<>();
+        for (final Serializable value : values) {
```

Comment:
stream().map() ?

Reply:
Cannot invoke stream() on the array type Serializable[]

Reply:
Arrays.stream()?

Reply:
Okay, that works. Thank Jan
`return Arrays.stream(values).map(value -> new ActualValue(value)).collect(Collectors.toList());`